Sources/BraintreePayPalNativeCheckout/BTPayPalNativeRequest.swift

```diff
@@ -6,17 +6,164 @@ import BraintreeCore
 import BraintreePayPal
 #endif
 
-protocol BTPayPalNativeRequest {
-    var hermesPath: String { get }
-    var paymentType: BTPayPalPaymentType { get }
+@objcMembers public class BTPayPalNativeRequest: NSObject {
 
-    func parameters(with configuration: BTConfiguration) -> [AnyHashable: Any]
-}
+    // MARK: - Public Properties
+    // next_major_version: subclass BTPayPalRequest once BraintreePayPal is in Swift.
+
+    /// Optional: The line items for this transaction. It can include up to 249 line items.
+    public var lineItems: [BTPayPalLineItem]?
+
+    /// Defaults to false. When set to true, the shipping address selector will be displayed.
+    public var isShippingAddressRequired: Bool
+
+    /// Optional: The merchant name displayed inside of the PayPal flow; defaults to the company name on your Braintree account
+    public var displayName: String?
+
+    ///  Optional: A locale code to use for the transaction.
+    ///  - Note: Supported locales are:
+    ///
+    /// `da_DK`,
+    /// `de_DE`,
+    /// `en_AU`,
+    /// `en_GB`,
+    /// `en_US`,
+    /// `es_ES`,
+    /// `es_XC`,
+    /// `fr_CA`,
+    /// `fr_FR`,
+    /// `fr_XC`,
+    /// `id_ID`,
+    /// `it_IT`,
+    /// `ja_JP`,
+    /// `ko_KR`,
+    /// `nl_NL`,
+    /// `no_NO`,
+    /// `pl_PL`,
+    /// `pt_BR`,
+    /// `pt_PT`,
+    /// `ru_RU`,
+    /// `sv_SE`,
+    /// `th_TH`,
+    /// `tr_TR`,
+    /// `zh_CN`,
+    /// `zh_HK`,
+    /// `zh_TW`,
+    /// `zh_XC`.
+    public var localeCode: String?
+
+    /// Defaults to false. Set to true to enable user editing of the shipping address.
+    /// - Note: Only applies when `shippingAddressOverride` is set.
+    public var isShippingAddressEditable: Bool
+
+    /// Optional: A valid shipping address to be displayed in the transaction flow. An error will occur if this address is not valid.
+    public var shippingAddressOverride: BTPostalAddress?
+
+    /// Optional: A risk correlation ID created with Set Transaction Context on your server.
+    public var riskCorrelationID: String?
+
+    /// Optional: A non-default merchant account to use for tokenization.
+    public var merchantAccountID: String?
+
+    // MARK: - Internal Properties
+
+    var hermesPath: String
+
+    var paymentType: BTPayPalPaymentType
+
+    // MARK: - Initializer
+
+    init(
```

Comment:
Needs docstring

Reply:
This init is internal. I know we've gone back and fourth on internal docstrings. Thoughts?

Reply:
Ah, yeh if it's not public I think its OK to omit!

Yeh I think in our style guide we _wanted_ to document everything including internal, but it ends up being a lot of work 😶

Reply:
🙈 agree